executor/show.go

```diff
@@ -196,6 +196,39 @@ func (e *ShowExec) fetchAll(ctx context.Context) error {
 	return nil
 }
 
+// visibleChecker checks if a stmt is visible for a certain user.
+type visibleChecker struct {
+	defaultDB string
+	ctx       sessionctx.Context
+	is        infoschema.InfoSchema
+	manager   privilege.Manager
+	ok        bool
+}
+
+func (v *visibleChecker) Enter(in ast.Node) (out ast.Node, skipChildren bool) {
+	switch x := in.(type) {
+	case *ast.TableName:
+		schema := x.Schema.L
+		if schema == "" {
+			schema = v.defaultDB
+		}
+		if !v.is.TableExists(model.NewCIStr(schema), x.Name) {
+			v.ok = false
```

Comment:
How about setting `v.ok` to `true` if table does not exist in the information schema?

Reply:
`v.ok=v.is.TableExists(model.NewCIStr(schema), x.Name)`?

Reply:
@eurekaka Why? If the table does not exist anymore, why should we show it?
@crazycs520 No, `ok` may change from fasle to true then, which should not happen.